spec/api/v1/external_users/claims/advocates/final_claim_spec.rb

```diff
@@ -41,35 +41,73 @@
-      post_to_validate_endpoint
-      [
-        'first_day_of_trial is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'trial_concluded_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'trial_fixed_notice_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'trial_fixed_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'trial_cracked_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'retrial_started_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'retrial_concluded_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
-        'main_hearing_date is not in an acceptable date format (YYYY-MM-DD[T00:00:00])'
-      ].each do |error|
-        expect(last_response.status).to eq(400)
-        expect(last_response.body).to include(error)
+  context 'when CLAIR contingency functionality is disabled' do
+    before { valid_params.except!(:main_hearing_date) }
+
+    include_examples 'advocate claim test setup'
+    it_behaves_like 'a claim endpoint', relative_endpoint: FINAL_CLAIM_ENDPOINT
+    it_behaves_like 'a claim validate endpoint', relative_endpoint: FINAL_CLAIM_ENDPOINT
+    it_behaves_like 'a claim create endpoint', relative_endpoint: FINAL_CLAIM_ENDPOINT
+
+    # TODO: write a generic date error handling spec and share
+    describe "POST #{ClaimApiEndpoints.for(FINAL_CLAIM_ENDPOINT).validate}" do
+      it 'returns 400 and JSON error when dates are not in acceptable format' do
+        valid_params[:first_day_of_trial] = '01-01-2015'
+        valid_params[:trial_concluded_at] = '09-01-2015'
+        valid_params[:trial_fixed_notice_at] = '01-01-2015'
+        valid_params[:trial_fixed_at] = '01-01-2015'
+        valid_params[:trial_cracked_at] = '01-01-2015'
+        valid_params[:retrial_started_at] = '01-01-2015'
+        valid_params[:retrial_concluded_at] = '01-01-2015'
+        post_to_validate_endpoint
+        [
+          'first_day_of_trial is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'trial_concluded_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'trial_fixed_notice_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'trial_fixed_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'trial_cracked_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'retrial_started_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])',
+          'retrial_concluded_at is not in an acceptable date format (YYYY-MM-DD[T00:00:00])'
+        ].each do |error|
+          expect(last_response.status).to eq(400)
+          expect(last_response.body).to include(error)
+        end
+      end
+    end
+  end
+
+  context 'when CLAIR contingency functionality is enabled',
+          skip: 'Skipped pending removal of the main_hearing_date feature flag' do
```

Comment:
Can `pending` be used instead of `skip`? A pending test will report as a failure if it starts passing so it alerts when the feature has been fixed (or turned on, in this case). I don't know if `pending` can be used as metadata on the `context`, like `skip` here.

Reply:
I'd have liked to use `pending` but there are issues with it:

* It's not possible to use `pending` on individual failing tests as they are mostly in shared examples (actually, all of them are following my refactor). Marking them as pending causes failures when they run in the context where the feature flag is disabled.

* It's possible to mark a whole `context` as pending, similar to how I used `skip` in the PR, but as not all the specs in that context are failing, using `pending` leads to failures where the tests are passing (if that makes sense 🤯).

If you've got any thoughts on other approaches I'd love to hear them 🙂

Reply:
OK. This should be quite short lived so it is probably not worth it.